File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& mat) {
        // Transpose
        for (int i = 0; i < mat.size(); i++) {
            for (int j = i; j < mat.size(); j++) {
                swap(mat[i][j], mat[j][i]);
            }
        }
        //reve
        for(auto& x : mat){
            reverse(x.begin() , x.end());
        }
    }
    bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
        for (int i = 0; i < 4; i++) {
            if (mat == target) {
                return true;
            }
            rotate(mat);
        }

        return false;
    }
};

File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.go
func rotate(mat [][]int){
    n := len(mat)

    //transpose 
    for i:= 0 ; i<n ; i++ {
        for j:= i ; j<n ; j++ {
            mat[i][j] , mat[j][i] = mat[j][i] , mat[i][j];
        }
    }
    //rev each row
    for i:= 0 ; i<n ; i++ {
        for j:= 0 ; j< n/2 ; j++ {
            k := n- j -1;
            mat[i][j] , mat[i][k] = mat[i][k] , mat[i][j];
        }
    }
}



func findRotation(matrix [][]int, target [][]int) bool {
   // n := len(matrix)

    for i:=0 ; i< 4 ; i++ {
        if equal(matrix , target) {
            return true;
        }
        rotate(matrix);

    }
    return false;

  
}


  func equal(a [][]int , b [][]int) bool {
        n := len(a);

        for i := 0 ; i< n ; i++{
            for j:=0 ; j<n ; j++{
                if(a[i][j] != b[i][j]){
                   return false;
                }
            }
        }

        return true;
    }